lib/application.js

```diff
@@ -179,117 +206,87 @@ Application.prototype.createClient = function () {
-      throw error
-    })
-  })
-
-  /**
-   * Utility from webdriverio v5
-   * https://github.com/webdriverio/webdriverio/blob/v5.9.4/packages/webdriverio/src/commands/browser/switchWindow.js
-   */
-  this.client.addCommand('switchWindow', async function (urlOrTitleToMatch) {
-    if (typeof urlOrTitleToMatch !== 'string' && !(urlOrTitleToMatch instanceof RegExp)) {
-      throw new TypeError('Invalid parameter urlOrTitleToMatch: expected a string or a RegExp')
-    }
-
-    const tabs = await this.windowHandles().then(getResponseValue)
-
-    for (const tab of tabs) {
-      await this.window(tab)
-
-      /**
-       * check if url matches
-       */
-      const url = await this.getUrl()
-      if (url.match(urlOrTitleToMatch)) {
-        return tab
-      }
-
-      /**
-       * check title
-       */
-      const title = await this.getTitle()
-      if (title.match(urlOrTitleToMatch)) {
-        return tab
-      }
-    }
+  this.client.addCommand('waitUntilTextExists', function (
+    selector,
+    text,
+    timeout
+  ) {
+    var self = this
+    return self
+      .waitUntil(async function () {
+        const elem = await self.$(selector)
+        const exists = await elem.isExisting()
+        if (!exists) {
+          return false
+        }
 
-    throw new Error(`No window found with title or url matching "${urlOrTitleToMatch}"`)
+        const selectorText = await elem.getText()
+        return Array.isArray(selectorText)
+          ? selectorText.some((s) => s.includes(text))
+          : selectorText.includes(text)
+      }, timeout)
+      .then(
+        function () {},
+        function (error) {
+          error.message = 'waitUntilTextExists ' + error.message
+          throw error
+        }
+      )
   })
 
   this.client.addCommand('waitUntilWindowLoaded', function (timeout) {
-    return this.waitUntil(function () {
-      return this.webContents.isLoading().then(function (loading) {
-        return !loading
-      })
-    }, timeout).then(function () { }, function (error) {
-      error.message = 'waitUntilWindowLoaded ' + error.message
-      throw error
-    })
+    var self = this
+    return self
+      .waitUntil(function () {
+        return self.webContents.isLoading().then(function (loading) {
+          return !loading
+        })
+      }, timeout)
+      .then(
+        function () {},
+        function (error) {
+          error.message = 'waitUntilWindowLoaded ' + error.message
+          throw error
+        }
+      )
   })
 
   this.client.addCommand('getWindowCount', function () {
-    return this.windowHandles().then(getResponseValue).then(function (handles) {
+    return this.getWindowHandles().then(function (handles) {
       return handles.length
     })
   })
 
   this.client.addCommand('windowByIndex', function (index) {
-    return this.windowHandles().then(getResponseValue).then(function (handles) {
-      return this.window(handles[index])
+    var self = this
```

Comment:
`self` shouldn't be needed here.

Reply:
Unfortunately this is necessary -- `TypeError: this.switchToWindow is not a function` errors if we don't include it